http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/SamDAO.scala

```diff
@@ -72,9 +72,10 @@ trait SamDAO[F[_]] {
     implicit ev: Ask[F, TraceId]
   ): F[Option[UserSubjectId]]
 
-  def getUserSubjectIdFromToken(token: String)(implicit ev: Ask[F, TraceId]): F[Option[UserSubjectId]]
+  def getSamUserInfo(token: String)(implicit ev: Ask[F, TraceId]): F[Option[SamUserInfo]]
```

Comment:
Nit: the pattern in this class seems to use the `Authorization` type for the token instead of `String`, but I don't feel that strongly.